pkg/resources/cruisecontrol/configmap.go

```diff
@@ -146,6 +146,10 @@ func GenerateCapacityConfig(kafkaCluster *v1beta1.KafkaCluster, log logr.Logger,
 	// If there is already a config added manually, use that one
 	if kafkaCluster.Spec.CruiseControlConfig.CapacityConfig != "" {
 		userProvidedCapacityConfig := kafkaCluster.Spec.CruiseControlConfig.CapacityConfig
+		updatedProvidedCapacityConfig, err := extendUserProvidedCapacityConfig([]byte(userProvidedCapacityConfig), kafkaCluster, log)
```

Comment:
I think the logic would be neater and easier to read:
1. if user provides cc config than check for broker id "-1", if present return original capacity config
2. if broker id "-1" is not present than append the missing capacity config
3. if there is no user provided capacity config just append the missing capacity config to an empty config

Reply:
This is exactly what's happening? Or I may not understand what you mean. First we check if there's a user provided config, if there is we enter `extendUserProvidedCapacityConfig` where first we check if there is a "-1" ID present and if there is, we return, if there isn't we add generated values as we agreed previously. After all this we're on the "other side" of the if checking if the user provided their own config and we go on to call `appendGeneratedBrokerCapacities`, which adds generated values to an empty config basically.